README.md

```diff
@@ -312,6 +312,33 @@ and a second layer containing the script only.
 In real life, the isolated layer can contains a Python environment or
 Node modules.
 
+Keep in mind that each layer will only be able to strip out nix store paths
+that exist in their _child_ layers (because layers can contain layers), not in
+their _sibling_ layers. So, if you did it like this, chances are that there
+were some repeated nix store paths between layers 1 and 2. In that case, you
+must choose between optimizing image cache or image size:
+
+```nix
+{ pkgs }:
+let
+  application = pkgs.writeScript "conversation" ''
+    ${pkgs.hello}/bin/hello
+    echo "Haaa aa... I'm dying!!!"
+  '';
+in
+pkgs.nix2container.buildImage {
+  name = "hello";
+  config = {
+    entrypoint = ["${pkgs.bash}/bin/bash" application];
+  };
+  layers = [
+    # These layers might duplicate a store path
+    (pkgs.nix2container.buildLayer { deps = [pkgs.bash]; })
```

Comment:
It could be nice to move these expressions to the let binding and name them `layer 1` / `layer 2` as explained in the above section.